web_ext/sseq_gui/tests/test_s_2.py

```diff
@@ -120,11 +120,17 @@ def test_history(driver):
 
     timeout = 0.1
     while True:
+        print(f"Waiting for {timeout} seconds")
         time.sleep(timeout)
 
         try:
             with open(f"{driver.tempdir}/s_2.save") as f:
-                driver.check_file("s_2.save", f.read())
+                file_contents = f.read()
+                if file_contents == "":
+                    # The driver hasn't finished saving the file yet, wait a bit longer
+                    print("Waiting for driver to finish")
+                    raise FileNotFoundError
+                driver.check_file("s_2.save", file_contents)
```

Comment:
What about:
```py
timeout = 0.1
save_path = Path(driver.tempdir) / "s_2.save"
while True:
   print(f"Waiting for {timeout} seconds")
   time.sleep(timeout)
   file_contents = save_path.read_text()
   if save_path.read_text():
      break
   timeout *= 2
driver.check_file("s_2.save", file_contents)
````

Reply:
what about:
```py
file_contents = save_path.read_text() if save_path.exists() else None
```

Reply:
I agree that's cleaner. Do we want to keep the max 10s for the timeout?

Reply:
Yeah probably, but was there a max of 10s for the timeout before? I don't see any indication of that. Maybe `timeout = min(2*timeout, 10)`?

Reply:
It was on line 131, just after the end of the diff unfortunately. I put it in the suggested changes commit